### PR TITLE
[codex] Add triggered rule composition primitives

### PR DESCRIPTION
**Pull request by TheCookieLab**
Fixes #.

Changes proposed in this pull request:
- Add `TriggeredRule` for staged rule composition where an always-active invalidation/timeout rule can coexist with trigger/delegate stages such as target-armed trailing exits. Existing boolean composites (`and`/`or`/`xor`/chain/vote) do not model armed lifecycle, activation windows, delegate priming, or reset policies, so this is the reusable rule primitive for that behavior.
- Harden stop-loss, stop-gain, trailing stop, open-position timeout, and wait-for rules with explicit validation, null/NaN guards, touch-or-beyond boundary coverage, and lower-allocation trailing stop evaluation.
- Extend rule serialization for boolean-array constructor parameters and document staged exit composition, reset choices, and touch-vs-cross rule selection in the README and changelog.

- [x] I have run `mvn -B clean license:format formatter:format test install` to format and test my changes per the [contributing guidelines](.github/CONTRIBUTING.md)
- [x] I have added an entry with applicable ticket number(s) to the appropriate unreleased section of `CHANGELOG.md`

## Release Notes

* **New Features**
  * Added `TriggeredRule` for composing multi-stage exit logic with configurable activation windows and reset policies.

* **Bug Fixes**
  * Enhanced null-safety and validation across stop-loss, stop-gain, and trailing stop rules.
  * Improved robustness of minimum bar count and wait-for rules.

* **Documentation**
  * Added staged exit rules guide with Java examples demonstrating composed exit strategies.

* **Tests**
  * Comprehensive test coverage for new and updated rule behaviors, including constructor validation and boundary conditions.

[Review Change Stack](https://app.coderabbit.ai/change-stack/ta4j/ta4j/pull/1530?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by TheCookieLab** [comment]
@coderabbitai full review

**Comment by TheCookieLab** [comment]
@copilot apply changes based on the comments in [this thread](https://github.com/ta4j/ta4j/pull/1530#pullrequestreview-4306054917)

File: ta4j-core/src/main/java/org/ta4j/core/rules/TriggeredRule.java [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+    private final boolean[] primeDelegateWhileInactive;
+    private final ResetPolicy[] resetPolicies;
+    private final transient boolean hasPositionResetStages;
+    private final transient boolean hasExplicitResetStages;
+    private final transient StageState[] stageStates;
+
+    private transient PositionSnapshot lastPositionSnapshot;
+    private transient TradingRecord lastTradingRecord;
+    private transient int lastEvaluatedIndex = Integer.MIN_VALUE;
+    private transient boolean evaluationStarted;
```
[the remark this answers is not in the document]
Comment (TheCookieLab) [reply ta4j-core/src/main/java/org/ta4j/core/rules/TriggeredRule.java:147]: @copilot apply changes based on [this feedback](https://github.com/ta4j/ta4j/pull/1530#discussion_r3255219621). make sure to run the full build at the end to verify and apply formatting
